src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -410,15 +413,16 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             throws IOException, InterruptedException {
         setEnv(build.getEnvironment(listener));
         final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
+        FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
         
         // Copy MATLAB scratch file into the workspace only if Automatic option is selected.
         if (testRunMode.contains(AUTOMATIC_OPTION)) {
-            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
-                    workspace, getClass().getClassLoader());
+            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE, targetWorkspace);
         }
         ProcStarter matlabLauncher;
         try {
-            MatlabReleaseInfo rel = new MatlabReleaseInfo(getLocalMatlab());
+            FilePath nodeSpecificMatlabRoot = new FilePath(launcher.getChannel(),getLocalMatlab());
+            MatlabReleaseInfo rel = new MatlabReleaseInfo(nodeSpecificMatlabRoot);
```

Comment:
since the only exception being caught is one coming from MatlabReleaseInfo, only this code needs to be in the try.

Reply:
The scope of Exception is widened now with the copy operation. and the intention is to catch any exception and exit with an appropriate exit status. changes the exception type accordingly .